src/api/retrieveQuizQuestions.js

```diff
@@ -0,0 +1,91 @@
+//const axios = require('axios').default;
+
+const questionsApis = {
+  DOGS: { url: 'https://api.thedogapi.com/v1/breeds', api_key: '62b8cc9a-2c13-4eec-abe2-80703eabb0a6' },
+  CATS: { url: 'https://api.thecatapi.com/v1/breeds', api_key: '2d4cf1ee-1883-474f-80ab-f931262fd79b' },
+};
+
+const makeAnimalArray = (data) => {
+  const questions = data.map((record) => {
+    const animal = {
+      name: record.name,
+      img: record.image,
+    };
+    return animal;
+  });
+  return questions;
+};
+
+const prepareQuiz = (arr, numberOfQuestions) => {
+  const questions = [];
+  for (let questionNumber = 1; questionNumber <= numberOfQuestions; questionNumber++) {
+    const animal = arr[Math.floor(Math.random() * arr.length)];
+    console.log('Question number: ', questionNumber, ' answer is: ', animal);
+    const incorrectAnswers = [];
+
+    arr.splice(arr.indexOf(animal), 1);
```

Comment:
In this case, you are mutating the original array. If there are 20 elements in the retrieved array, at some point, there will be no elements to create incorrectAnswers from.

Reply:
I did that on purpose, by mutating the array I can ensure, that there would not be any duplication. Also only one pet is removed from the array, falses remain in the array.